packages/account/tests/account.e2e.spec.ts

```diff
@@ -477,4 +476,246 @@ describe("Account Tests", () => {
+  }, 60000);
+
+  it("should withdraw nativeToken and an erc20 token", async () => {
+    const usdt = "0xda5289fcaaf71d52a80a254da614a192b693e977";
+
+    const {
+      whale: { viemWallet: signer, publicAddress: smartAccountOwner },
+      bundlerUrl,
+      publicClient,
+      biconomyPaymasterApiKey,
+    } = mumbai;
+
+    const smartAccount = await createSmartAccountClient({
+      signer,
+      biconomyPaymasterApiKey,
+      bundlerUrl,
+    });
+
+    const smartAccountAddress = await smartAccount.getAddress();
+    const balanceOfSABefore = (await checkBalance(publicClient, smartAccountAddress)) as bigint;
+    const balanceOfRecipientBefore = (await checkBalance(publicClient, smartAccountOwner)) as bigint;
+    const usdtBalanceOfSABefore = await checkBalance(publicClient, smartAccountAddress, usdt);
+    const usdtBalanceOfRecipientBefore = await checkBalance(publicClient, smartAccountOwner, usdt);
+
+    const { wait } = await smartAccount.withdraw(
+      smartAccountOwner,
+      [
+        { address: usdt, amount: BigInt(1) },
+        { address: NATIVE_TOKEN_ALIAS, amount: BigInt(1) },
+      ],
+      {
+        paymasterServiceData: { mode: PaymasterMode.SPONSORED },
+      },
+    );
+
+    const {
+      receipt: { transactionHash },
+      userOpHash,
+      success,
+    } = await wait();
+
+    expect(userOpHash).toBeTruthy();
+    expect(success).toBe("true");
+    expect(transactionHash).toBeTruthy();
+
+    const balanceOfSAAfter = (await checkBalance(publicClient, smartAccountAddress)) as bigint;
+    const balanceOfRecipientAfter = (await checkBalance(publicClient, smartAccountOwner)) as bigint;
+    const usdtBalanceOfSAAfter = (await checkBalance(publicClient, smartAccountAddress, usdt)) as bigint;
+    const usdtBalanceOfRecipientAfter = (await checkBalance(publicClient, smartAccountOwner, usdt)) as bigint;
+
+    expect(balanceOfSABefore - balanceOfSAAfter).toBe(1n);
+    expect(balanceOfRecipientAfter - balanceOfRecipientBefore).toBe(1n);
+    expect(usdtBalanceOfSAAfter - usdtBalanceOfSABefore).toBe(-1n);
+    expect(usdtBalanceOfRecipientAfter - usdtBalanceOfRecipientBefore).toBe(1n);
+  }, 60000);
+
+  it("should withdraw all native token", async () => {
+    const {
+      whale: { viemWallet: signer, publicAddress: smartAccountOwner, account },
+      bundlerUrl,
+      viemChain,
+      publicClient,
+      biconomyPaymasterApiKey,
+    } = mumbai;
+
+    const smartAccount = await createSmartAccountClient({
+      signer,
+      biconomyPaymasterApiKey,
+      bundlerUrl,
+    });
+
+    const smartAccountAddress = await smartAccount.getAddress();
+    const balanceOfSABefore = (await checkBalance(publicClient, smartAccountAddress)) as bigint;
+    const balanceOfRecipientBefore = (await checkBalance(publicClient, smartAccountOwner)) as bigint;
+
+    const { wait } = await smartAccount.withdraw(smartAccountOwner, undefined /* null or undefined or [] */, {
```

Comment:
how do you choose the amount to withdraw if paymaster is not involved?

if a paymaster call fails, we would withdraw all which will fail on chain again

Reply:
They could call the getBalances method :)

Reply:
Can you elaborate on this second point? I don't follow

Reply:
if I want to withdraw native tokens and max amount (say 1 Matic) what's there in SA 
and if there is no paymaster or paymaster call fails then pnd goes '0x' which means account payas for gas in native tokens. so we can't transfer 1 and also pay 0.01 matic in gas

Reply:
Yes that's right. I have thrown [this error](https://github.com/bcnmy/biconomy-client-sdk/pull/424/files/fecb066bce1f444f4537e5bdf518ceddac40d2d2#diff-78e7afa7e87e267fae298f597794c5ab45544d36cd675d4dd9aa843337bb652bR337) if a user does not provide his own amount in this case. dev will be expected to find his own amounts to withdraw if no paymaster is used. It'll be on the dev to find the right amount

Reply:
https://github.com/bcnmy/biconomy-client-sdk/pull/424/files#diff-78e7afa7e87e267fae298f597794c5ab45544d36cd675d4dd9aa843337bb652bR349

Reply:
I'm basically telling the dev that it's up to them to pick the amount of dust they want to leave behind, assuming they're not using a paymaster.